src/shout_subgroup/group_chat_listener.py

```diff
@@ -88,41 +88,46 @@ async def listen_for_messages_handler(update: Update, context: ContextTypes.DEFA
         last_name=update.message.from_user.last_name
     )
 
-    session = get_database()
-    try:
-        maybe_added_user = await add_user_to_group_chat(session, update.effective_chat, user_who_sent_the_message)
-        # TODO: Add message like "The bot has recognized John Doe"
-        return
-    except NotGroupChatError:
-        # await update.message.reply_text("Sorry, you can only create or modify subgroups in group chats.")
-        return
+    db_session = get_database()
+
+    with db_session.begin() as session:
```

Comment:
Also, I'm using a pattern I found in [FastAPI documentation](https://fastapi.tiangolo.com/tutorial/sql-databases/?h=data#create-a-dependency). The pattern is to create a db session for each API request (command handlers in our case), complete the transaction, then close the db session